sqlglot/dialects/snowflake.py

```diff
@@ -236,6 +237,19 @@ class Snowflake(Dialect):
         "ff6": "%f",
     }
 
+    @classmethod
+    def quote_identifier(cls, expression: E, identify: bool = True) -> E:
+        # This disables quoting DUAL in SELECT ... FROM DUAL, because Snowflake treats an
+        # unquoted DUAL keyword in a special way and does not map it to a user-defined table
+        if (
+            isinstance(expression, exp.Identifier)
+            and isinstance(expression.parent, exp.Table)
+            and expression.name.lower() == "dual"
```

Comment:
what if it's quoted

Reply:
It will remain quoted

Reply:
why do you call lower though

Reply:
because you can have both upper and lower case versions of the unquoted version and so you gotta normalize to compare

Reply:
if you match anything which is `dual` when lowercased then you don't wanna touch it, if it's quoted then fine but if not you can't proceed calling `Dialect.quote_identifier` on it because you risk messing up the query